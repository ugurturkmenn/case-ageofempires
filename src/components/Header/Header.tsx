import { Link } from "react-router-dom";

const Header = () => {
  return (
    <div>
      <header className='text-gray-700 body-font border-b border-gray-200'>
        <div className='container mx-auto flex flex-wrap p-5 flex-col md:flex-row items-center'>
          <nav className='md:ml-auto flex flex-wrap items-center text-base justify-center gap-5'>
            <Link to={'/'}>Home</Link>
            <Link to={'/units'}>Units</Link>
          </nav>
        </div>
      </header>
    </div>
  )
}

export default Header
