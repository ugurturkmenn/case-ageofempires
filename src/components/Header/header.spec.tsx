import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Header from "./Header";
import { ReactNode } from "react";

const renderWithRouter = (ui: ReactNode) => {
  return render(
    <BrowserRouter>
      {ui}
    </BrowserRouter>
  );
};

describe('Header Component', () => {

  test('should match the snapshot', () => {
    const { asFragment } = renderWithRouter(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render the Home and Units links', () => {
    renderWithRouter(<Header />);

    const homeLink = screen.getByText(/Home/i);
    const unitsLink = screen.getByText(/Units/i);

    expect(homeLink).toBeInTheDocument();
    expect(unitsLink).toBeInTheDocument();
  });

  test('should navigate to the correct route when links are clicked', () => {
    renderWithRouter(<Header />);

    const homeLink = screen.getByText(/Home/i);
    const unitsLink = screen.getByText(/Units/i);

    expect(homeLink.getAttribute('href')).toBe('/');
    expect(unitsLink.getAttribute('href')).toBe('/units');

    fireEvent.click(homeLink);
    fireEvent.click(unitsLink);
  });

});
