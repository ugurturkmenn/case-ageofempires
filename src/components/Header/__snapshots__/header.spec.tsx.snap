// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <header
      class="text-gray-700 body-font border-b border-gray-200"
    >
      <div
        class="container mx-auto flex flex-wrap p-5 flex-col md:flex-row items-center"
      >
        <nav
          class="md:ml-auto flex flex-wrap items-center text-base justify-center gap-5"
        >
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/units"
          >
            Units
          </a>
        </nav>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
